src/registry/domain/s3.js

```diff
@@ -177,17 +171,17 @@ module.exports = function(conf) {
     if (fileInfo.gzip) {
       obj.ContentEncoding = 'gzip';
     }
-
-    getClient().putObject(obj, callback);
+    const upload = getClient().upload(obj);
+    upload.send(callback);
```

Comment:
For large files, the recommended approach is to use upload over putObject. Upload automatically processes the req into multiple chunks and its `Body` property accepts both string (making this change retrocompatible) and streams (see change in the `putFile` method).